docs/2-fork-accountability.md

```diff
@@ -0,0 +1,191 @@
+
+## The problem
+
+A fork is a split in the blockchain network or, with other words, a divergence into different block paths. 
+A **fork** happens when two correct validators decide on different blocks in the same height (i.e., there are two commits for different blocks in the same consensus instance). 
+
+In the context of the Tendermint protocol, a fork happens in height *h* when a quorum of messages (*2f + 1* PRECOMMIT messages) are sent for different values: more formally, there are two sets of *2f + 1* PRECOMMIT messages, A and B, such that A's messages are PRECOMMIT votes for value *v* and B's messages are PRECOMMIT votes for value *v' != v* in height *h*.  
+
+The Tendermint conensus algorithm guarantees that a fork (i.e., a violation of the agreement property) cannot happen when the number of faulty processes is less than one third.
+We want to address and analyze the problem of defining a Tendermint consensus algorithm that provides the same four properties (validity, agreement, termination, integrity) when the number of faulty processes is less than one third and guarantees, in addition, an efficient detection of faulty processes that can potentially make the consensus fail with a fork when there are more than one third of faulty validators.
+
+In particular, we assume there are more than one third of faulty validators such that the following holds:
+
+    f < num. faulty processes <= 2f
+
+In the Tendermint protocol, all processes should be accountable for their actions and faulty validators should be identified and punished according to the protocol specifications, for example by removing them from the validators set.
+We aim to give incentives to validators to behave correctly according to the protocol specifications and detect faulty validators, without mistakenly detecting correct validators. 
+ 
+## Misbehaviors
+ 
+In the previous section, we mentioned that a process can make the consensus fail with a fork. There are, indeed, multiple reasons that can generate a fork in the blockchain, all coming from processes that deviate from the protocol specification in various ways.
+
+We can summarize some of the most important misbehaviors that can possibly cause a fork in a consensus instance according to the Tendermint consensus algorithm rules: 
+
+- A process sends multiple proposals in a round *r* for different values 
+
+- A process sends a PREVOTE/PRECOMMIT message for an invalid value *v* in a round *r*
+
+- A process sends multiple PREVOTE/PRECOMMIT messages in a round *r* for different values (**equivocation**)
+
+- A process sends a PREVOTE/PRECOMMIT message for a value *v* despite having a lock on a different value *v'*
+
+- A process sends a PRECOMMIT for a value *v* without having received at least two thirds PREVOTE messages for *v* 
+
+Processes are considered correct if they do not make any of the above-mentioned misbehaviors.
```

Comment:
I would remove this sentence. There are undetectable ways for processes to be faulty and they are not considered correct in the standard literature if they commit those.

Reply:
You're right, I was mentioning this in the scope of the accountability project though. Removed.